File: python/trading-api/get_corporate_action_history.py
#!/usr/bin/env python3
import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(__file__)))
from dnse import DNSEClient

def main():
    client = DNSEClient(
        api_key="replace-with-api-key",
        api_secret="replace-with-api-secret",
        base_url="https://openapi.dnse.com.vn",
    )

    status, body = client.get_corporate_action_history(
        account_no="0001000115",
        symbol="SSI",
        ca_type="all",
        ca_status="pending",
        page_index=0,
        page_size=20,
        dry_run=False,
    )
    print(status, body)


if __name__ == "__main__":
    main()

File: javascript/examples/get_corporate_action_history.js
'use strict';

const { DNSEClient } = require('../dnse');

async function main() {
  const client = new DNSEClient({
    apiKey: process.env.DNSE_API_KEY || 'replace-with-api-key',
    apiSecret: process.env.DNSE_API_SECRET || 'replace-with-api-secret',
    baseUrl: process.env.DNSE_BASE_URL || 'https://openapi.dnse.com.vn',
  });

  const { status, body } = await client.getCorporateActionHistory('0001234567', {
    fromDate: '2026-01-01',
    toDate: '2026-01-31',
    dryRun: false,
  });

  console.log(status, body);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
